ddev/ddev
# feat: set Drupal database to mariadb default on config update, fixes #6160

## The Issue

- #6160

## How This PR Solves The Issue

Updates `ddev config --update`:
Sets the default database type if there is no database volume, otherwise displays a warning message to let users know that they may want to migrate their database.

## Manual Testing Instructions

```
mkdir my-drupal-site && cd my-drupal-site
ddev config --project-type=drupal --php-version=8.3 --docroot=web --database=mariadb:10.4
ddev start
ddev composer create drupal/recommended-project:^11.x-dev
ddev config --update
```

See the warning:

```
Default database type is mariadb:10.11, but the current actual database type is mariadb:10.4, you may want to migrate with 'ddev debug migrate-database mariadb:10.11'.
```

Then:

```
ddev delete -Oy
ddev config --update
```

See:

```
Configuring drupal project with database type 'mariadb:10.11'
```

## Automated Testing Overview



## Related Issue Link(s)

## Release/Deployment Notes
